### PR TITLE
[WIP] feat(components) Add filterable metadata to components

### PR DESCRIPTION
**Pull request by awm086**

### CODE REVIEW

File: packages/bodiless-ui/src/ReactTags.css [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+ *           <mark />
+ *         </li>
+ *         <li class="is-disabled">
+ *           <mark />
+ *         </li>
+ *       </ul>
+ *     </div>
+ *   </div>
+ */
+
```
[the remark this answers is not in the document]
Comment (VancheeZze) [reply packages/bodiless-ui/src/ReactTags.css:39]: @wodenx  The only problem with the tailwind styling is that ReactTags does not provide an option to specify classes for the suggestions list and list items - it only has a class for the suggestions wrapper. With the tailwind, we won't be able to target <ul> or <li> elements ( see template example on lines 28-35 ). There is an option though to create a custom tailwind component ( in @bodiless/ui bodiless.tailwind.config ) either forReactTags suggestions or for the ReactTags in general.

File: examples/test-site/src/data/pages/filter-item/index.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const suggestions = [
+  { id: 3, name: 'Bananas' },
+  { id: 4, name: 'Mangos' },
+  { id: 5, name: 'Lemons' },
+  { id: 6, name: 'Apricots' },
+  { id: 7, name: 'not' },
+];
+
+const TagableItem = flow(asBodilessFilterItem())('span');
```
[the remark this answers is not in the document]
Comment (awm086) [reply examples/test-site/src/data/pages/filter-item/index.tsx:32]: @wodenx so I can just remove the ('span')

File: packages/bodiless-components/src/Taggable/Item.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    const {
+      ComponentFormTitle,
+      ComponentFormText,
+      ComponentFormUnwrapButton,
+      ReactTags,
+    } = getUI(formUi);
+    return (
+      <>
+        <ComponentFormTitle>Group Membership</ComponentFormTitle>
+        <ComponentFormText type="hidden" field="tags" />
```
[the remark this answers is not in the document]
Comment (awm086) [reply packages/bodiless-components/src/Taggable/Item.tsx:83]: I am not sure how do that when we are getting the components  from formUi.

File: packages/bodiless-core/src/contextMenuForm.tsx [hunk 4]
```diff
@@ -16,6 +16,7 @@ import React, { ReactNode } from 'react';
 import {
   Form, FormApi, FormState, Text,
 } from 'informed';
+import ReactTagsBase from 'react-tag-autocomplete';
```
[the remark this answers is not in the document]
Comment (awm086) [reply packages/bodiless-core/src/contextMenuForm.tsx:19]: @VancheeZze thoughts?

**Pull request closed without merging**